Assignment 2.Rmd

```diff
@@ -95,6 +122,51 @@ Now practice with data from our class. Please create a **person-network** with t
 
 Once you have done this, also [look up](http://igraph.org/r/) how to generate the following network metrics: betweeness centrality and dregree. **Who is the most central person in the network?**
 
+```{r}
+library(tidyr)
+library(dplyr)
+library(tibble)
+library(stringr)
+library(igraph)
+library(reshape2)
+
+# Import data
+table <- read.table("hudk4050-classes.txt", header = TRUE, sep = ",")
+
+# Crop data
+classdata <- table[-c(1:2),c(20:26)]
+person_class <- gather(classdata, "QID", "CID", 2:7)
+person_class <- person_class[,-2]
+person_class <- na_if(person_class, "")
+person_class <- na.omit(person_class)
+
+# Drop incomplete data
+person_class <- subset(person_class, str_detect(person_class[,2], "[A-Z]+\\s*[0-9]+"))
+
+# Format data
+person_class[,2] <- gsub("([A-Z]+)\\s*([0-9]+)", "\\1 \\2", person_class[,2])
+
+# Modify ambiguous data
+person_class[person_class == 'QMSS-G 5072' | person_class == 'G 5072'] <- 'QMSS 5072'
+person_class[person_class == 'QMSS G 5015'] <- 'QMSS 5015'
+person_class[person_class == 'QMSS GR 5067' | person_class == 'G 5067'] <- 'QMSS 5067'
+
+person_person <- merge(person_class, person_class, by = 'CID')
+names(person_person) <- c('CID', 'FROM', 'TO')
+person_person <- subset(person_person, !FROM == TO)
+
+person_person.edge <- count(person_person, FROM, TO)
+names(person_person.edge) <- c('FROM', 'TO', 'count')
+
+person_person.matrix <- pivot_wider(person_person.edge, names_from = FROM, values_from = count)
+person_person.matrix %>% remove_rownames %>% column_to_rownames(var='TO')
+
+person_person.graph <- graph_from_data_frame(person_person.edge, directed=FALSE)
+plot(person_person.graph, layout=layout.sphere, vertex.label = V(person_person.graph)$FROM, vertex.size = 1)
```

Comment:
Good job! Very beautiful data wrangling. Because I can not run your code here, I am not sure if you create a right person_person matrix, since you merged the same data frames together to make it. We can further discuss that if you would like to.